src/modules/rangeproof/rangeproof_impl.h

```diff
@@ -202,88 +301,12 @@ SECP256K1_INLINE static int secp256k1_rangeproof_genrand(secp256k1_scalar *sec,
     return ret;
 }
 
-SECP256K1_INLINE static int secp256k1_range_proveparams(uint64_t *v, size_t *rings, size_t *rsizes, size_t *npub, size_t *secidx, uint64_t *min_value,
- int *mantissa, uint64_t *scale,  int *exp, int *min_bits, uint64_t value) {
-    size_t i;
-    *rings = 1;
-    rsizes[0] = 1;
-    secidx[0] = 0;
-    *scale = 1;
-    *mantissa = 0;
-    *npub = 0;
-    if (*min_value == UINT64_MAX) {
-        /* If the minimum value is the maximal representable value, then we cannot code a range. */
-        *exp = -1;
-    }
-    if (*exp >= 0) {
-        int max_bits;
-        uint64_t v2;
-        if ((*min_value && value > INT64_MAX) || (value && *min_value >= INT64_MAX)) {
-            /* If either value or min_value is >= 2^63-1 then the other must by zero to avoid overflowing the proven range. */
-            return 0;
```

Comment:
I think the old check was just wrong, in multiple ways.
* First, if `min_value` is greater than `INT64_MAX` or indeed greater than 0, then `value` must also be greater than 0, since it must be >= `min_value`. In the old code this was checked before even calling `proveparams`.
* If this instead intended to check that `min_value == value` in the case that either one was > `INT64_MAX`, this still makes no sense. There is no reason I shouldn't be able to have `min_value` be `INT64_MAX + 1` and `value` be `INT64_MAX + 2`, say.